app/summarize.js

```diff
@@ -1,26 +1,40 @@
+var _ = require('underscore');
+
 module.exports = function(event) {
   var errors = [];
+  // Deep-clone the event plan
   var summary = JSON.parse(JSON.stringify(event.plan));
+
   summary.forEach(function(f, i){
+    // Plan and snapshot offsets may not match, get the correct key
+    var fkey = _.findKey(event.snapshot, function (feature) {
```

Comment:
Can you clarify when these don't match? I've definitely misunderstood something here.